File: two_sum.py
def twoSum(arr, target):
  # left pointer
    left = 0
  # right pointer
    right = len(arr)-1
    while left < right:
  # find the sum of the two pointers, then run through the ifs
      current = arr[left] + arr[right]
       if current < target:
       # increase left pointer
          left += 1
        elif current > target:
       # increase right pointer
          right -= 1
       # else means that current and target are equal; current is the sum of the left and right pointers
        else:
          return
            [left, right]
    return [-1, -1]

  # enter example data here
    arr1 = []
    target1 =
    arr2 = []
    target2 =
    arr2 = []
    target3 =

    print(twoSum(arr1, target1))
    print(twoSum(arr2, target2))
    print(twoSum(arr3, target3))

File: two_sum.cpp
#include <vector>
using namespace std;

class Solution
{
public:
  vector<int> twoSum(vector<int> &data, int target)
  {
    int left = 0;
    int right = data.size() - 1;
    while (left < right)
    {
      int sum = data[left] + data[right];
      if (sum == target)
        break; // dont need to return here since we return at the end of the while loop
      else if (sum < target)
        right--;
      else if (sum > target)
        left++;
      else
        return data;
    }
    return {left, right};
  }
};